include/core/Layer.hpp

```diff
@@ -102,7 +103,11 @@ namespace ngen
         /***
          * @brief Run one simulation timestep for each model in this layer
         */
-        virtual void update_models()
+        virtual void update_models(boost::span<double> catchment_results, 
```

Comment:
Being a bit picky about naming, let's call these `catchment_outflows` and `nexus_flows`. Ideally, someone who has a better handle on the hydrology can check me on these being good names. They're at least less generic than `results`